### DISCUSSION TITLE
Backup directory size no where near size diagnostic reports , attachments not copied at all

### DISCUSSION THREAD

**Discussion by fxfactorial** [ReagentX/imessage-exporter#293] [Q&A] [answered]
Great tool - thank you 

iOS on 17.5.1 created unencrypted backup on macOS 14.5. iMessage diagnostic says 

```
Attachment diagnostic data:
    Total attachments: 4989
        Data referenced in table: 39.96 GB
        Data present on disk: 36.75 GB
    Missing files: 167 (3%)
        No path provided: 0
        No file located: 167
```

but directory created by iMessage-exporter contains no attachments directory & its total size is only 6.3M. No errors, no warnings. 

No idea why the attachments aren't coming over from the backup

**Comment by ReagentX** [comment]
What command did you run to generate the export?

**Reply by fxfactorial** [reply]
```
$ imessage-exporter -f txt -p ~/Library/Application\ Support/MobileSync/Backup/00008110-000158422692801E -a iOS -o backup_export
```

**Reply by ReagentX** [reply, marked as answer]
You need to specify `--copy-method` to include attachments:

- [Instructions](https://github.com/ReagentX/imessage-exporter/tree/develop/imessage-exporter#how-to-use)
- [Examples](https://github.com/ReagentX/imessage-exporter/tree/develop/imessage-exporter#examples)

**Reply by fxfactorial** [reply]
Ah perfect - that was it thank you. 

two things: 

1. It deviates from macOS backup which I did find attachments even without --copy-method given. 
2. Efficient sounds like some processing happening anyway but efficient actually means copy as is.

**Reply by ReagentX** [reply]
1. That is not accurate, `--copy-method` is always [disabled by default](https://github.com/ReagentX/imessage-exporter/blob/9862a2b46afbeb8ad63043c02060e0fe7acd156f/imessage-exporter/src/app/options.rs#L218-L226). 
3. `efficient` does not perform any processing on attachment copies; `compatible` converts attachments to be more compatible (i.e. web-friendly).
